tests/models/test_cpu.py

```diff
@@ -75,57 +70,85 @@ def test_cpu_slurm_save_load(tmpdir):
     # new logger file to get meta
     logger = tutils.get_default_logger(tmpdir, version=version)
 
+    model = BoringModel()
+
+    class _StartCallback(Callback):
+        def on_init_start(self, trainer):
+            print("Starting to init trainer!")
+
+        # set the epoch start hook so we can predict before the model does the full training
+        def on_epoch_start(self, trainer, model):
+            assert trainer.global_step == real_global_step and trainer.global_step > 0
+            # predict with loaded model to make sure answers are the same
+            model.eval()
+            new_pred = model(batch)
+            assert torch.all(torch.eq(pred_before_saving, new_pred)).item() == 1
```

Comment:
```suggestion
            mode = model.training
            model.eval()
            new_pred = model(batch)
            assert torch.eq(pred_before_saving, new_pred).all()
            model.train(mode)
```
should be put back to the training mode.